asm/instruction.go

```diff
@@ -106,6 +134,26 @@ func (ins Instruction) Marshal(w io.Writer, bo binary.ByteOrder) (uint64, error)
 		return 0, fmt.Errorf("can't marshal registers: %s", err)
 	}
 
+	if ins.OpCode.Class().IsALU() {
+		switch ins.OpCode.ALUOp() {
```

Comment:
Should we return an error if Offset is not 0 here? It would indicate that someone is trying to mix extended opcodes with manually specifiying offset.

Reply:
Not sure how strict we want to be on parsing. It was sort of neat that we basically handled the ISAv4 extension already, it was just the long jump rewriting that would not have worked, but we did not error on unknown instructions. I think we should be strict on the generation side but permissive on the parsing side

Reply:
> It was sort of neat that we basically handled the ISAv4 extension already, [...] we did not error on unknown instructions.

agreed

> I think we should be strict on the generation side but permissive on the parsing side

Not sure what you mean. I would like the following:

1. Round tripping unknown opcodes should work. Worst case you don't get pretty printing.
2. Using the `asm` package to assemble things should generate useful errors if possible.

Since we do `Offset = 0` in `Unmarshal` we know that roundtripping will work since it can't hit the `Offset != 0` case. So if we return an error here we'd make (2) possible.

Reply:
> Since we do Offset = 0 in Unmarshal we know that roundtripping will work since it can't hit the Offset != 0 case. 

Right. My mind was still on Offset set to whatever we got. Yes, then it should be fine.